torch/_inductor/fx_passes/post_grad.py

```diff
@@ -47,6 +48,91 @@
 inference_patterns = PatternMatcherPass()
 
 
+def get_storage(t):
+    return t.untyped_storage()._cdata
+
+
+def get_node_storage(node):
+    if not isinstance(node.meta["val"], torch.Tensor):
+        return None
+    if not torch._C._has_storage(node.meta["val"]):
+        return None
+    return get_storage(node.meta["val"])
+
+
+class FakeTensorUpdater:
+    def __init__(self, graph):
+        self.processed_hashes = set()
+        self.graph = graph
+
+        for node in self.graph.nodes:
+            self.processed_hashes.add(self.hash_node(node))
+
+    def hash_node(self, node):
+        return (node, node.target, node.args, node.kwargs)
+
+    def incremental_update(self):
+        """
+        Not fully featured yet. Mainly added to ensure that reinplacing pass is
+        sound.
+
+        Todo things:
+        1. We probably want to persist this across more passes.
+        """
+        return
+        processed = set()
+        existing_tensors = defaultdict(int)
+        for node in self.graph.nodes:
+            existing_tensors[get_node_storage(node)] += 1
+
+        def is_fake_tensor_same(new, old):
+            if new.shape != old.shape or new.stride() != old.stride():
+                return False
+            if get_storage(new) == get_storage(old):
+                return True
+            if (
+                existing_tensors[get_storage(old)] == 1
+                and get_storage(new) not in existing_tensors
+            ):
+                return True
+            return False
+
+        for node in self.graph.nodes:
```

Comment:
It would be good to have a more "incremental" version of `incremental_update` that doesn't require a pass over the entire graph, since if we make lots of small changes this will be O(n^2).

Reply:
Hmm... I'm not totally sure how we can do this without more significant changes to FX to record changed nodes every time. Any ideas?

Reply:
Currently, whenever we add nodes they are all contiguous.  So you could pass in a "seed" node (the newly added node) then just expand from the seed.